aten/src/ATen/native/Unique.cpp

```diff
@@ -92,7 +92,7 @@ std::tuple<Tensor, Tensor, Tensor> unique_consecutive_cpu_template(
     inverse_data = inverse_indices.data_ptr<int64_t>();
   }
   if (return_counts) {
-    counts.resize_(input.sizes());
+    counts.resize_({numel});
```

Comment:
I feel that checking for numel > 0 on line 83 and just returning empty output, inverse_indices and counts there instead of having a few checks in the function body will be simpler and more readable.

Reply:
fixed